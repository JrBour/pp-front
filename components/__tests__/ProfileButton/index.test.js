import { shallowMount } from '@vue/test-utils'
import ProfileButton from '@/components/ProfileButton'

describe('ProfileButton', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(ProfileButton, {
      propsData: {
        icon: 'lock',
        text: 'Edit password'
      },
    })
  })

  describe('methods', () => {
    it('should emit handle-click when we click on button', async () => {
      const button = wrapper.find('button')
      await button.trigger('click')
      expect(wrapper.emitted('handle-click')).toBeTruthy()
    })
  })

  describe('snapshot', () => {
    it('renders properly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
