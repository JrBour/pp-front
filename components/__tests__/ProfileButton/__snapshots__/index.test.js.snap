// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileButton snapshot renders properly 1`] = `
<button><img src="" alt="lock" class="profile__icon">
  <div class="profile__button_title">
    <p>Edit password</p>
  </div> <img src="" alt="Red arrow">
</button>
`;
